test/Dialect/Struct/structs_fail.llzk

```diff
@@ -40,7 +40,7 @@ module attributes {veridise.lang = "llzk"} {
 }
 // -----
 module attributes {veridise.lang = "llzk"} {
-  // expected-error@+1 {{'struct.def' op must define exactly one 'constrain' function}}
+  // expected-error@+1 {{'struct.def' op must define only "@compute" and "@constrain" functions, or a "@product" function; missing "@constrain" or "@product"}}
```

Comment:
@raghav198 can you also add a test that hits the `missing "@compute" or "@product"` case,

Reply:
Yeah, that makes sense. It's fine to leave the checks for duplicates here. I think go ahead and add the tests with multiple of each function and expect the "redefinition" error. These will act as regression tests.